feat(server): add full attestation support to RSK MCP Server with new…

Ad-Capital opened the pull request:
… tools (issue, verify, revoke, list, create-schema), RAS integration, Zod validation, and updated docs

Ad-Capital committed:
feat(server): add full attestation support to RSK MCP Server with new tools (issue, verify, revoke, list, create-schema), RAS integration, Zod validation, and updated docs

Ad-Capital committed:
refactor(schemas): rename createSchemaSchema to createSchema.

Remove redundant "Schema" suffix from createSchemaSchema export to improve readability.

Ad-Capital commented:
Heyy... I’ve applied the changes based on your feedback. Please take a look
and let me know if further adjustments are required.

On Mon, Dec 1, 2025 at 10:20 AM Sebas ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In package.json
> <https://github.com/rsksmart/rsk-mcp-server/pull/81#discussion_r2576271196>
> :
>
> > @@ -38,7 +41,7 @@
>    ],
>    "dependencies": {
>      ***@***.***/sdk": "^1.13.2",
> -    ***@***.***/rsk-cli": "^1.3.1",
> +    ***@***.***/rsk-cli": "^1.4.0",
>
> Please for now, do not change this information, we will change it at the
> end.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rsksmart/rsk-mcp-server/pull/81#pullrequestreview-3523986943>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7QZ75KOCVAYGPHFSWE2HVL37QB7DAVCNFSM6AAAAACLK5F7O6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTKMRTHE4DMOJUGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

scguaquetam commented:
Hello @Ad-Capital , now that the another PR on the CLI is clearer, you can start implementing the changes now in the MCP server, you need to start by implementing correctly implementation of the attestation command on `src/services/AttestationService`  first line, when trying to import attestation service from the CLI 
`import { attestationCommand } from "@rsksmart/rsk-cli/dist/src/commands/attestation.js";`


<img width="1003" height="231" alt="image" src="https://github.com/user-attachments/assets/1975e306-8de4-4a10-837b-37b260e92ac9" />

Ad-Capital committed:
feat(attestations): add full attestation support with RAS integration

- Add AttestationService with issue, verify, revoke, list, and create-schema methods
- Integrate with RSK CLI attestation utilities (attestation.js, walletSigner.js)
- Add required dependencies: @ethereum-attestation-service/eas-sdk, ethers
- Update package.json with attestation dependencies
- Add attestation schemas and types for MCP tools
- Add GraphQL support for querying attestations via EASScan endpoint

Ad-Capital committed:
feat(attestations): add full attestation support with RAS integration

- Add AttestationService with issue, verify, revoke, list, and create-schema methods
- Integrate with RSK CLI attestation utilities (attestation.js, walletSigner.js)
- Add required dependencies: @ethereum-attestation-service/eas-sdk, ethers
- Update package.json with attestation dependencies
- Add attestation schemas and types for MCP tools
- Add GraphQL support for querying attestations via EASScan endpoint

Ad-Capital commented:
Hi @scguaquetam, I’ve addressed all the feedback: fixed the lockfile version, restored the CLI dependency, and implemented AttestationService using existing CLI utilities. All five attestation features are complete and working, pending the CLI utility release. Ready for review.

Ad-Capital committed:
fix(attestations): refactor attestation service and fix response handling

- Remove @ts-ignore hacks in AttestationService by initializing EAS directly
  via RSK_ATTESTATION_CONFIG instead of accessing private CLI service internals
- Add staticNetwork: true and explicit chain IDs (30/31) to JsonRpcProvider
  to fix ethers.js v6 network detection timeout on RSK nodes
- Replace GraphQL list-attestations with eth_getLogs event query; add rpcUrl
  param with actionable error for RSK public node limitation
- Add attest-deployment, attest-verification, attest-transfer tools using
  rsk-cli createDeploymentAttestation/createVerificationAttestation/createTransferAttestation
- Validate DEFAULT_SCHEMA_UIDS upfront in specialized attest tools to give
  accurate errors instead of generic null returns
- Fix returnAttestationVerifiedSuccessfully to use data.uid and data.status
  (was showing undefined UID and ACTIVE for non-existent attestations)
- Fix isError flag in provideResponse so MCP error responses are correctly
  signalled to clients (was always returning success at transport level)
- Fix all viewer URLs from rootstock.easscan.org to explorer.rootstock.io/ras
- Add DEFAULT_SCHEMA_UIDS to rsk-cli.d.ts ambient declarations
- Add integration test script covering read-only and full write lifecycle

Ad-Capital committed:
docs(attestations): update README with new tools, correct explorer URLs, and accurate project structure

- Add attest-deployment, attest-verification, attest-transfer tool documentation
- Add rpcUrl param and RSK public node limitation note to list-attestations
- Fix explorer URLs from explorer.rsk.co to explorer.rootstock.io
- Fix project structure: types.d.ts → rsk-cli.d.ts, add scripts/ directory
- Update @rsksmart/rsk-cli dependency version

Ad-Capital committed:
Merge branch 'main' into coPilot

scguaquetam committed:
fix/vulnerabilities by npm audit fix

scguaquetam committed:
fix/ running npm audit and deleting unused code

ezequiel-rodriguez reviewed (approved):
LGTM
